src/algebra/squarefree.lean

```diff
@@ -204,4 +204,71 @@ lemma sum_divisors_filter_squarefree {n : ℕ} (h0 : n ≠ 0)
 by rw [finset.sum_eq_multiset_sum, divisors_filter_squarefree h0, multiset.map_map,
     finset.sum_eq_multiset_sum]
 
+lemma sq_mul_squarefree {n : ℕ} (hn : 0 < n) :
+  ∃ a b : ℕ, 0 < a ∧ 0 < b ∧ b ^ 2 * a = n ∧ squarefree a :=
+begin
+  set S := {s ∈ finset.range (n + 1) | s ∣ n ∧ ∃ x, s = x ^ 2 } with hS,
```

Comment:
so in these `let` works just fine as you never `rw hS`; `simp` can rewrite `let`s by itself so you could replace this with `let` and `simp [hS]` with `simp [S]`. hopefully Scott has an answer how to use `let` in rewrites in the other issue! (it may be a case of `erw`, maybe?)